sdk/python/test/e2e/test_e2e_mpijob.py

```diff
@@ -68,39 +65,39 @@ def test_sdk_e2e_with_gang_scheduling():
         )),
     )
 
-    mpijob = generate_mpijob(launcher, worker, V1SchedulingPolicy(min_available=10))
-    patched_mpijob = generate_mpijob(launcher, worker, V1SchedulingPolicy(min_available=2))
+    mpijob = generate_mpijob(launcher, worker, V1SchedulingPolicy(min_available=10), job_namespace)
+    patched_mpijob = generate_mpijob(launcher, worker, V1SchedulingPolicy(min_available=2), job_namespace)
 
-    TRAINING_CLIENT.create_mpijob(mpijob, JOB_NAMESPACE)
+    training_client.create_mpijob(mpijob, job_namespace)
```

Comment:
Should we create a `job_namespace` if there isn't a namespace in the cluster before creating a CustomJob resource?

Reply:
yes, that makes sense.

Reply:
Actually, we should delete all the resources that are created as part of testing. For new namespaces, it's okay as deleting those namespaces at the end of testing will not cause any issues. But if someone is using a pre-existing namespace for testing, then deleting that namespace can cause issues. We can update tests to delete the namespace if it's created as part of testing but that might complicate test code.

And also since users are providing the namespace as input, they should be responsible for creating it.

I think it's best to leave it to the user.
What do you think?

Reply:
@nagar-ajay This makes sense to me. I feel, it is better to avoid complications of namespace deletion.
 @tenzen-y Isn't it better to assume that namespace is already created?

Reply:
It a lot of makes sense. I agree with @nagar-ajay.